docs/USAGE-GUIDE.md

```diff
@@ -1585,6 +1585,102 @@ int s2n_config_add_ticket_crypto_key(struct s2n_config *config, const uint8_t *n
+
+The folllowing APIs enable the caller to obtain the pre-shared key that was selected by the server for the connection. 
+
+**s2n_connection_get_negotiated_psk_identity_length** gets the negotiated pre-shared key identity length from the s2n connection object. If the negotiated pre-shared key does not exist, the value `0` is returned.  
+
+**s2n_connection_get_negotiated_psk_identity** gets the negotiated pre-shared key identity from the s2n connection object. If the negotiated pre-shared key does not exist, the PSK identity will not be obtained and no error will be returned.  
+
+### Configuring Client Offered Pre-Shared Keys
+ 
+The folllowing APIs enable the client to configure the list of pre-shared keys supported.
+
+**s2n_offered_psk** creates a new s2n offered pre-shared key object. An offered pre-shared key object represents a single pre-shared key (PSK) shared by the client. This object is used to create the list of offered pre-shared keys shared by the client. Use `s2n_offered_psk_free` to free the memory allocated to the s2n offered pre-shared key object created by this API. 
+
+**s2n_offered_psk_free** frees the memory associated with the `s2n_offered_psk` object.
+
+### Using the Offered Pre-Shared Keys to Select a Pre-Shared Key
+
+The following APIs enable the client to use it's list of offered pre-shared keys to set the callback that selects the matching server pre-shared key to be used for the connection. Use the `s2n_config_set_psk_selection_callback` only when a custom pre-shared key selection logic is desired by the client. By default in the pre-shared key exension, s2n-tls server chooses the first identity in its pre-shared key list that also appears in the client's pre-shared key list.
+
+**s2n_offered_psk_get_identity** gets the pre-shared key identity and pre-shared key identity length for a given offered pre-shared key object. Callers can use this API to obtain the identity value set on the `s2n_offered_psk` object.
+
+**s2n_offered_psk_list_has_next** checks whether the offered pre-shared key list has an offered psk object next in line in the list. An offered pre-shared key list contains all the pre-shared keys offered by the client for the server to select.
+If an offered pre-shared key exists next in line, this API returns True, otherwise False. Use this API prior to calling `s2n_offered_psk_list_next` to obtain the next offered PSK object from the offered PSK list. 
+
+**s2n_offered_psk_list_next** obtains the next offered PSK object from the offered PSK list. 
+Use `s2n_offered_psk_list_has_next` prior to this API call to determine if an offered PSK exists next in the list. 
+
+**s2n_offered_psk_list_reread** resets the offered pre-shared key list pointer to the initial pre-shared key object in the order in which the pre-shared key objects were originally added to the list by the client. Use this API along with the `s2n_psk_selection_callback` callback, if needed, to select a pre-shared key for the connection.
```

Comment:
An example would probably help, but it might also help to discuss these three together and how they interact. That's the benefit of the usage guide over the doxygen comments.